sklearn/linear_model/tests/test_least_angle.py

```diff
@@ -428,6 +434,24 @@ def test_no_warning_for_zero_mse():
     assert_true(np.any(np.isinf(lars.criterion_)))
 
 
+def test_lars_path_readonly_data():
```

Comment:
you could probably do a more light-weight test by using stringIO and np.memmap? I don't like IO in tests.... but seems good enough.